example/bfvmm_main/src/main.cpp

```diff
@@ -0,0 +1,43 @@
+//
```

Comment:
I moved bfvmm_main to be an example project. This provides an example of all three override points that this PR adds for extension writters. Details:

* vmm_init gets called on a single core only, and the domain that gets passed as an argument has fully contructed vcpus for each core on the user's system. One key note about this function is that it does NOT run in vmx-root.
* vmm_main gets called on each core in vmx-root as the first thing that happens immediately after the vcpu is loaded (triggered by a special CPUID leaf, just like how it worked before).
* vmm_fini gets called in vmx-root just before destruction is about to happen.

When you take all three of these functions together, there should be no more need for vcpu inheritance of any kind.